### DISCUSSION TITLE
FalconPy API development environment

### DISCUSSION THREAD

**Discussion by framontb** [CrowdStrike/falconpy#1170] [Q&A]
Are there a way to test FalconPy scripts against a development environment ?
Don't know if there are develpment environments where you can possibly create fake client_id and client_secret to test FalconPy scritps.

**Comment by jshcodes** [comment]
Hi @framontb -

Yes there is!  Unit tests can be found in the `tests` folder within this repo. In order to execute _every_ test, you will need to provide an API client ID / secret that is configured to allow all scopes (at a minimum of READ).

You specify these values using the environment variables:

- DEBUG_API_ID
- DEBUG_API_SECRET

If you wish to enable full debug output, also set the following environment variable to the value of `DEBUG`.

- FALCONPY_UNIT_TEST_DEBUG

At this point you should be able to run individual service collection tests or the full series.

To run an individual unit test (from the root folder):
```shell
util/unit-test.sh firewall_management
```

or

```shell
util/unit-test.sh hosts
```

etc.

To run the full series (also from the root folder):

```shell
util/run-tests.sh
```

Thank you for the question! 🙇
